### PR TITLE
test: fix Python unittests

### PR DESCRIPTION
**Pull request by cclauss**
Add __pytest__ to our Travis CI jobs to run 27 Python tests.

##### Checklist

- [ ] `npm install && npm test` passes
- [x] tests are included 
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by cclauss** [comment]
@joaocgreis Can you please take a look at the failing Windows tests?  A fix did not jump out at me.

File: gyp/pylib/gyp/MSVSSettings_test.py [hunk 1]
```diff
@@ -1085,6 +1085,7 @@ def testConvertToMSBuildSettings_full_synthetic(self):
             'GenerateManifest': 'true',
             'IgnoreImportLibrary': 'true',
             'LinkIncremental': 'false'}}
+    self.maxDiff = 9999  # on failure display a long diff
```
Comment (bnoordhuis) [inline gyp/pylib/gyp/MSVSSettings_test.py]: `self.maxDiff = None` if you don't really want an upper limit?

Comment (cclauss) [reply gyp/pylib/gyp/MSVSSettings_test.py]: Understood.  These are reasonable limits for these test results.

**Comment by richardlau** [comment]
> 
> 
> @joaocgreis Can you please take a look at the failing Windows tests? A fix did not jump out at me.

I think part of the issue is that Travis has Visual Studio 2017 on Windows and the version of `gyp` in this repository doesn't detect that because we have the VS detection in JavaScript (see https://github.com/nodejs/node-gyp/pull/1952#issuecomment-549631126) outside of `gyp`.

**Comment by joaocgreis** [comment]
@cclauss pushed a fix for the issues on Windows.

There is still a failure with Python 3.5 that looks like a real issue.

**Comment by joaocgreis** [comment]
Travis had failed on my repo earlier, but passed on a new run. Possibly some caching issue or something that got fixed. I guess this is good then!

**Comment by rvagg** [comment]
04da736

**Pull request closed without merging**